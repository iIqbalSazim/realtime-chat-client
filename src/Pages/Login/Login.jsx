import { Link } from "react-router-dom";

import LoginForm from "./Form/LoginForm";

const Login = () => {
  return (
    <main className="h-screen flex-col w-screen flex justify-center items-center">
      <LoginForm />
      <div>
        Don&apos;t have an account?{" "}
        <Link to={"/register"} className="btn-link">
          Register
        </Link>
      </div>
    </main>
  );
};

export default Login;
