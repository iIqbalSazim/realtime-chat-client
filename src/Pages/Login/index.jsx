import { Link } from "react-router-dom";
import LoginForm from "./Form/LoginForm";

const Login = () => {
  return (
    <main className="h-screen flex-col w-screen flex justify-center items-center">
      <LoginForm />
      <Link to={"/1"} className="btn btn-active btn-link">
        Go to Dashboard
      </Link>
    </main>
  );
};

export default Login;
