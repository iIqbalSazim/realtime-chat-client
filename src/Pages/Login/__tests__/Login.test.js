import { screen } from "@testing-library/react";

import { renderWithProviders } from "../../../../jest.setup";

import Login from "../Login";

describe("Login", () => {
  test("renders correctly", () => {
    const { container } = renderWithProviders(<Login />);

    const dontHaveAnAccount = screen.getByText(/Don't have an account?/i);
    const registerLink = screen.getByRole("link", { name: /register/i });

    expect(dontHaveAnAccount).toBeInTheDocument();
    expect(registerLink).toBeInTheDocument();
    expect(registerLink).toHaveAttribute("href", "/register");

    expect(container).toMatchSnapshot();
  });
});
