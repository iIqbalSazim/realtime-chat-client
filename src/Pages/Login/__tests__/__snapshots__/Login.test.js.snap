// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<div>
  <main
    class="h-screen flex-col w-screen flex justify-center items-center"
  >
    <form
      class="flex flex-col gap-4 items-center justify-center w-[500px] px-10 py-12 border border-slate-200"
    >
      <h1
        class="font-medium text-lg mb-5"
      >
        LOGIN
      </h1>
      <div
        class="form-control w-full max-w-xs"
      >
        <label
          class="label"
        >
          <span
            class="label-text"
          >
            Username
          </span>
        </label>
        <input
          class="input input-bordered w-full max-w-xs"
          name="username"
          placeholder="Your username"
          type="text"
        />
      </div>
      <div
        class="form-control w-full max-w-xs"
      >
        <label
          class="label"
        >
          <span
            class="label-text"
          >
            Password
          </span>
        </label>
        <input
          class="input input-bordered w-full max-w-xs"
          name="password"
          placeholder="Your password"
          type="password"
        />
      </div>
      <button
        class="btn btn-md btn-secondary font-semibold mt-5"
        type="submit"
      >
        Submit
      </button>
    </form>
    <div>
      Don't have an account?
       
      <a
        class="btn-link"
        href="/register"
      >
        Register
      </a>
    </div>
  </main>
</div>
`;
